dbt_coverage/__init__.py

```diff
@@ -317,40 +317,34 @@ def to_json(self):
 class CoverageDiff:
     """Dataclass summarizing the difference of two coverage reports, mainly its decrease."""
 
-    before: CoverageReport
+    before: Optional[CoverageReport]
     after: CoverageReport
     new_misses: Dict[str, CoverageDiff] = field(init=False)
 
-    # @formatter:off
     def __post_init__(self):
-        assert self.before.cov_type == self.after.cov_type, \
+        assert self.before is None or self.before.cov_type == self.after.cov_type, \
             f"Cannot compare reports with different cov_types: {self.before.cov_type} and " \
             f"{self.after.cov_type}"
-        assert self.before.report_type == self.after.report_type, \
+        assert self.before is None or self.before.report_type == self.after.report_type, \
             f"Cannot compare reports with different report_types: {self.before.report_type} and " \
             f"{self.after.report_type}"
 
         self.new_misses = self.find_new_misses()
-    # @formatter:on
 
     def find_new_misses(self):
         if self.after.report_type == CoverageReport.Type.COLUMN:
             return None
 
-        new_misses_names = self.after.misses - self.before.misses \
-            if self.before.misses is not None \
-            else self.after.misses
+        new_misses_names = self.after.misses - (self.before.misses if self.before is not None
+                                                else set())
         new_misses_entity_names = set(miss.split('.', maxsplit=1)[0] for miss in new_misses_names)
 
         res: Dict[str, CoverageDiff] = {}
         for new_miss_entity_name in new_misses_entity_names:
-            res[new_miss_entity_name] = CoverageDiff(
-                self.before.subentities.get(new_miss_entity_name,
-                                            CoverageReport(self.before.report_type,
-                                                           self.before.cov_type,
-                                                           None, None, None, {})),
-                self.after.subentities[new_miss_entity_name]
-            )
+            before_entity = self.before.subentities.get(new_miss_entity_name) \
+                if self.before is not None else None
+            after_entity = self.after.subentities[new_miss_entity_name]
+            res[new_miss_entity_name] = CoverageDiff(before_entity, after_entity)
```

Comment:
The most important change happens here. Instead of creating a dummy coverage report to pass instead of a previously nonexistent entity (happens in the case of a newly added column or a table), we pass `None`.

Reply:
@sweco why was the dummy coverage report in there previously?

Reply:
I thought it would be a good idea to not have `None`s there, but in the end, it seems that it was much more complicated and less readable that way. A nice code is only done through rounds of refactoring iterations, this PR is one of them. 😊